lib/wasix/src/runtime/module_cache/types.rs

```diff
@@ -123,19 +124,22 @@ impl CacheError {
 
 /// The SHA-256 hash of a WebAssembly module.
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct ModuleHash([u8; 32]);
+pub enum ModuleHash {
+    Sha256([u8; 32]),
+    Xxhash([u8; 8]),
+}
 
 impl ModuleHash {
     /// Create a new [`ModuleHash`] from the raw SHA-256 hash.
     pub fn from_bytes(key: [u8; 32]) -> Self {
-        ModuleHash(key)
+        ModuleHash::Sha256(key)
     }
```

Comment:
The first commit was a partial implementation. I wasn't sure of the consequences that changing the hashing algorithm would have since places other than `load_module` use the `sha256` function. I wanted to see whether tests will pass or not. Now I've fully changed `ModuleCache` to use `xxhash` instead of `sha256` (the call to `sha256` will forward to `xxhash` instead). If all tests pass, my next commit will fully migrate to `xxhash`.

Reply:
I think allowing multiple hashing algorithms needs a little more careful planning. Just using an `enum` and encoding the algorithm using the upper 32bit of a number may not be the most optimal way. Also it's outside of the scope of this PR. I can create an issue for it though. What do you think?

Reply:
Fair. Please create an issue then 👍